src/components/checkbox/checkbox/checkbox.ts

```diff
@@ -1,71 +1,32 @@
-import {
-  type CSSResultGroup,
-  html,
-  LitElement,
-  type PropertyValues,
-  type TemplateResult,
-} from 'lit';
-import { customElement, property, state } from 'lit/decorators.js';
+import { LitElement, html, type CSSResultGroup, type TemplateResult } from 'lit';
+import { customElement, property } from 'lit/decorators.js';
 
-import { SbbLanguageController, SbbSlotStateController } from '../../core/controllers/index.js';
+import { SbbSlotStateController } from '../../core/controllers/index.js';
 import { EventEmitter } from '../../core/eventing/index.js';
-import { i18nCollapsed, i18nExpanded } from '../../core/i18n/index.js';
-import type {
-  SbbCheckedStateChange,
-  SbbDisabledStateChange,
-  SbbIconPlacement,
-  SbbStateChange,
-} from '../../core/interfaces/index.js';
-import {
-  SbbFormAssociatedCheckboxMixin,
-  SbbHydrationMixin,
-  SbbUpdateSchedulerMixin,
-} from '../../core/mixins/index.js';
-import { SbbIconNameMixin } from '../../icon/index.js';
-import type { SbbSelectionPanelElement } from '../../selection-panel/index.js';
-import type { SbbCheckboxGroupElement } from '../checkbox-group/index.js';
-
-import style from './checkbox.scss?lit&inline';
+import { SbbCheckboxCommonElementMixin, checkboxStyle, commonStyle } from '../common/index.js';
 
 import '../../screen-reader-only/index.js';
 import '../../visual-checkbox/index.js';
 
-export type SbbCheckboxStateChange = Extract<
-  SbbStateChange,
-  SbbDisabledStateChange | SbbCheckedStateChange
->;
-
 export type SbbCheckboxSize = 's' | 'm';
 
 /**
  * It displays a checkbox enhanced with the SBB Design.
  *
  * @slot - Use the unnamed slot to add content to the `sbb-checkbox`.
  * @slot icon - Slot used to render the checkbox icon (disabled inside a selection panel).
- * @slot subtext - Slot used to render a subtext under the label (only visible within a selection panel).
- * @slot suffix - Slot used to render additional content after the label (only visible within a selection panel).
  * @event {CustomEvent<void>} didChange - Deprecated. used for React. Will probably be removed once React 19 is available.
  * @event {Event} change - Event fired on change.
  * @event {InputEvent} input - Event fired on input.
  */
 @customElement('sbb-checkbox')
-export class SbbCheckboxElement extends SbbUpdateSchedulerMixin(
-  SbbFormAssociatedCheckboxMixin(SbbIconNameMixin(SbbHydrationMixin(LitElement))),
-) {
-  public static override styles: CSSResultGroup = style;
+export class SbbCheckboxElement extends SbbCheckboxCommonElementMixin(LitElement) {
```

Comment:
`events` are shared with the panel (which has 1 more); can it be moved in the common file?
same for the eventemitters (eg. _checkboxLoaded)